porcupine/plugins/filemanager.py

```diff
@@ -25,51 +25,84 @@
 else:
     trash_name = "trash"
 
+copy_path: Path | None = None
+
 
 def find_tabs_by_parent_path(path: Path) -> list[tabs.FileTab]:
     return [
         tab
         for tab in get_tab_manager().tabs()
         if isinstance(tab, tabs.FileTab)
-        and tab.path is not None
-        and (path == tab.path or path in tab.path.parents)
+           and tab.path is not None
+           and (path == tab.path or path in tab.path.parents)
     ]
 
 
-def ask_name_for_renaming(old_path: Path) -> Path | None:
+def ask_name_for_renaming(old_path: Path, is_paste=False) -> Path | None:
     label_width = 400
 
     dialog = tkinter.Toplevel()
     dialog.transient(get_main_window())
     dialog.resizable(False, False)
-    dialog.title("Rename")
 
     big_frame = ttk.Frame(dialog, padding=10)
     big_frame.pack(fill="both", expand=True)
-    ttk.Label(
-        big_frame, text=f"Enter a new name for {old_path.name}:", wraplength=label_width
-    ).pack(fill="x")
 
+    if is_paste and old_path.parent != copy_path.parent:
+        dialog.title("File conflict")
+        ttk.Label(
+            big_frame, text=f"{old_path.parent} already has a file named {old_path.name}."
+                            f"\nDo you want to overwrite it?",
+            wraplength=label_width
+        ).pack(fill="x")
+    else:
+        dialog.title("Rename")
```

Comment:
The title gets set to "Rename" when I paste a file into the same directory as where it is. This isn't great because I'm renaming, not copying. I think we need a third case for pastes within the same parent directory.